server/scripts/import-tests/index.js

```diff
@@ -346,58 +433,179 @@ const getTests = ({ builtDirectoryPath, testPlanVersionId, ats, gitSha }) => {
+                        target: {
+                            at: collected.target.at,
+                            referencePage: collected.target.referencePage,
+                            setupScript: collected.target.setupScript
+                        },
+                        instructions: {
+                            instructions: collected.instructions.instructions,
+                            mode: collected.instructions.mode
+                        },
+                        commands,
+                        assertions: collected.assertions.map(
+                            ({
+                                assertionId,
+                                priority,
+                                assertionStatement,
+                                assertionPhrase,
+                                refIds,
+                                tokenizedAssertionStatements
+                            }) => {
+                                const tokenizedAssertionStatement =
+                                    tokenizedAssertionStatements[
+                                        collected.target.at.key
+                                    ];
+
+                                return {
+                                    assertionId,
+                                    priority,
+                                    assertionStatement:
+                                        tokenizedAssertionStatement ||
+                                        assertionStatement,
+                                    assertionPhrase,
+                                    refIds
+                                };
+                            }
+                        )
+                    },
+                    renderedUrl,
+                    scenarios,
+                    assertions: collected.assertions.map((assertion, index) => {
+                        let priority = '';
+                        if (assertion.priority === 1) priority = 'MUST';
+                        if (assertion.priority === 2) priority = 'SHOULD';
+                        if (assertion.priority === 3) priority = 'MAY';
+
+                        const tokenizedAssertionStatement =
+                            assertion.tokenizedAssertionStatements[
+                                collected.target.at.key
+                            ];
+
+                        return {
+                            id: createAssertionId(testId, index),
+                            priority,
+                            assertionStatement:
+                                tokenizedAssertionStatement ||
+                                assertion.assertionStatement,
+                            assertionPhrase: assertion.assertionPhrase
+                        };
+                    }),
+                    viewers: []
+                };
+
+                tests.push(test);
+            }
+        } else {
+            const common = allCollected[0];
```

Comment:
This block reads quite densely with quite a bit of logic inside of the object construction. Not sure if breaking up into smaller functions would help but just wanted to note that it was challenging to read this.

Reply:
Fair point. I can address this feedback.

Reply:
@stalgiag I've pushed this change

Reply:
Much easier to read. Thank you!